src/server/generic_family.cc

```diff
@@ -1509,6 +1509,32 @@ OpStatus GenericFamily::OpMove(const OpArgs& op_args, string_view key, DbIndex t
   return OpStatus::OK;
 }
 
+void GenericFamily::RandomKey(CmdArgList args, ConnectionContext* cntx) {
+  const static size_t kNumEntries = 15;
+  const static size_t kMaxEntriesPerShard = 2;
+
+  StringVec keys;
+  keys.reserve(kNumEntries);  // garantees that vector won't scale up on another thread
+  ScanOpts scan_opts;
+  scan_opts.limit = kMaxEntriesPerShard;
+  DbContext db_cntx{.db_index = cntx->conn_state.db_index};
+
+  shard_set->RunBriefInParallel(
+      [&](EngineShard* shard) {
+        uint64_t cursor = 0u;
+        OpScan({shard, 0u, db_cntx}, scan_opts, &cursor, &keys);
```

Comment:
This isn't really random, it scans the beginning of the table, no?

Reply:
@romange can you think of a way to select a random entry (or bucket) from dash table? Maybe `+=` on an iterator with some random size? What would be the upper size though?

Reply:
I will try another approach.